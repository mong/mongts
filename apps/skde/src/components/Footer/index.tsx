import Image from "next/image";
import { imgLoader } from "qmongjs";
import Link from "next/link";
import Paper from "@mui/material/Paper";
import { styled } from "@mui/material/styles";
import Box from "@mui/material/Box";
import Grid from "@mui/material/Grid";
import { Stack } from "@mui/material";
import { ArrowLink } from "../ArrowLink";

const FooterWrapper = styled(Paper)(() => ({
  marginTop: 20,
  color: "white",
}));

const FooterGridTop = styled(Grid)(() => ({
  justifyContent: "space-around",
  background: "#333333",
}));

const FooterGridBottom = styled(Grid)(() => ({
  justifyContent: "space-around",
  background: "#1A1A1A",
}));

const StyledLink = styled(Link)(() => ({
  color: "#c4dbf3",
  textDecoration: "underline",
}));

export type FooterProps = {
  page: "behandlingskvalitet" | "helseatlas" | "sykehusprofil";
};

export const Footer = (props: FooterProps) => {
  const { page } = props;

  const leftMargin = 0;
  const topMargin = 2;

  return (
    <FooterWrapper>
      <FooterGridTop
        container
        columns={{ xs: 4, sm: 8, md: 12, lg: 12 }}
        className="footer"
        marginTop={0}
        marginBottom={0}
        spacing={4}
      >
        <Grid
          item
          xs={4}
          sm={8}
          md={3}
          lg={3}
          marginTop={topMargin}
          marginLeft={leftMargin}
          marginBottom={2}
        >
          <Stack spacing={3}>
            <h4>OM NETTSTEDET</h4>
            <ArrowLink
              href={"https://www.skde.no/om-skde/personvern/"}
              text="Personvern"
            />
            <ArrowLink
              href={"https://www.skde.no/om-skde/informasjonskapsler/"}
              text="Informasjonskapsler"
            />
            <ArrowLink
              href={
                "https://uustatus.no/nb/erklaringer/publisert/589a8d23-4993-446e-b7eb-ef310bfe1dd7"
              }
              text="Tilgjengelighetserklæring"
              diagonalArrow={true}
            />
          </Stack>
        </Grid>

        <Grid
          item
          xs={4}
          sm={8}
          md={3}
          lg={3}
          marginTop={topMargin}
          marginLeft={leftMargin}
        >
          <Stack spacing={3}>
            <h4>KONTAKT</h4>
            <ArrowLink
              href={"https://www.skde.no/om-skde/kontaktinformasjon/"}
              text="Kontakt SKDE"
              diagonalArrow={false}
              button={false}
            />
          </Stack>
        </Grid>

        <Grid item xs={0} sm={0} md={3} lg={3}></Grid>
        <Grid item xs={0} sm={0} md={3} lg={3}></Grid>
        <Grid item xs={4} sm={8} md={12} lg={12}></Grid>
      </FooterGridTop>

      <FooterGridBottom
        container
        columns={{ xs: 4, sm: 8, md: 12, lg: 12 }}
        className="footer"
        marginTop={0}
        spacing={4}
      >
        <Grid item xs={4} sm={8} md={12} lg={12}>
          <Link href={"https://www.skde.no/"}>
            <Image
              className="footer-logo"
              id="skde-footer-logo"
              loader={imgLoader}
              src={"/img/logos/logo-skde-neg.svg"}
              alt="SKDE-logo"
              width={129}
              height={52}
            />
          </Link>
        </Grid>

        <Grid item xs={4} sm={8} md={12} lg={12}>
          Senter for klinisk dokumentasjon og evaluering (SKDE) er en enhet i
          Helse Nord.
        </Grid>

        <Grid item xs={4} sm={4} md={4} lg={4}>
          <Stack>
            <b>TELEFON</b>
            <StyledLink href={"tel:77755800"}>77 75 58 00</StyledLink>
            <br />
            <b>EPOST</b>
            <StyledLink href={"mailto:postmottak@helse-nord.no"}>
              postmottak@helse-nord.no
            </StyledLink>
          </Stack>
        </Grid>

        <Grid item xs={4} sm={4} md={4} lg={4}>
          <Stack>
            <b>BESØKSADRESSE</b>
            <text>Sykehusvegen 23</text>
            <text> 9019 TROMSØ</text>

            <StyledLink href={"https://maps.app.goo.gl/ohLzsYb8v6YvEDfL9"}>
              Vis kart
            </StyledLink>
          </Stack>
        </Grid>

        <Grid item xs={4} sm={4} md={4} lg={4}>
          <Stack>
            <b>ORGANISASJONSNUMMER</b>
            <text>990803765{"\n"}</text>
            <Box marginTop={6}>
              <Link href={"https://www.helse-nord.no/"}>
                <Image
                  className="footer-logo"
                  id="helse-nord-logo"
                  loader={imgLoader}
                  src={"/img/logos/logo-helse-nord-neg.svg"}
                  alt="Helse Nord logo"
                  width={220}
                  height={76}
                />
              </Link>
            </Box>
          </Stack>
        </Grid>
        {page === "helseatlas" ? (
          <>
            <Grid item xs={4} sm={4} md={8} lg={8}>
              <Link href="https://helse-forde.no/" title="Link til Helse Førde">
                <Image
                  loader={imgLoader}
                  src={`/helseatlas/img/logos/helse-forde-hvit.svg`}
                  height={52}
                  width={(180 * 52) / 40}
                  alt="Helse Førde logo"
                />
              </Link>
            </Grid>
            <Grid item xs={4} sm={4} md={4} lg={4}>
              <Link href="https://helse-vest.no/" title="Link til Helse Vest">
                <Image
                  loader={imgLoader}
                  src={`/helseatlas/img/logos/helse-vest-hvit.svg`}
                  height={52}
                  width={(180 * 52) / 40}
                  alt="Helse Vest logo"
                />
              </Link>
            </Grid>
          </>
        ) : null}
        <Grid item xs={4} sm={4} md={0} lg={0}></Grid>
      </FooterGridBottom>
    </FooterWrapper>
  );
};
