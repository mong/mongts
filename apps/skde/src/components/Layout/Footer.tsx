import Link from "next/link";
import Image from "next/image";

import classNames from "./Footer.module.css";
import { imgLoader } from "../../helpers/functions";

export const Footer: React.FC<{}> = ({}) => {
  return (
    <footer className={classNames.footerContainer}>
      <div className={classNames.footerContent}>
        <div className={classNames.footerTop}>
          <strong>Senter for klinisk dokumentasjon og evaluering (SKDE)</strong>
          <br />
          Postboks 6
          <br />
          9038 Tromsø
          <br />
          <br />
          <strong>Helse Førde</strong>
          <br />
          Postboks 1000
          <br />
          6807 Førde
        </div>

        <div className={classNames.footerBottom}>
          <div className={classNames.logo}>
            <Link href="https://helse-nord.no/">
              <a title="Link til Helse Nord">
                <Image
                  loader={imgLoader}
                  src={`/helseatlas/img/logos/helse-nord-hvit.svg`}
                  height="40px"
                  width="180px"
                  alt="Helse Nord logo"
                />
              </a>
            </Link>
          </div>
          <div className={classNames.logo}>
            <Link href="https://www.skde.no/">
              <a title="Link til SKDE">
                <Image
                  loader={imgLoader}
                  src={`/helseatlas/img/logos/skde-hvit.svg`}
                  height="40px"
                  width="180px"
                  alt="SKDE logo"
                />
              </a>
            </Link>
          </div>
          <div className={classNames.logo}>
            <Link href="https://helse-forde.no/">
              <a title="Link til Helse Førde">
                <Image
                  loader={imgLoader}
                  src={`/helseatlas/img/logos/helse-forde-hvit.svg`}
                  height="40px"
                  width="180px"
                  alt="Helse Førde logo"
                />
              </a>
            </Link>
          </div>
          <div className={classNames.logo}>
            <Link href="https://helse-vest.no/">
              <a title="Link til Helse Vest">
                <Image
                  loader={imgLoader}
                  src={`/helseatlas/img/logos/helse-vest-hvit.svg`}
                  height="40px"
                  width="180px"
                  alt="Helse Vest logo"
                />
              </a>
            </Link>
          </div>
        </div>
      </div>
    </footer>
  );
};
