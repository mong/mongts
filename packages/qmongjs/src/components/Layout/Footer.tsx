import React from "react";
import Link from "next/link";
import skdeLogo from "../../img/logos/SKDE_hvit_lys.png";
import helseNordLogo from "../../img/logos/hf_nord-white.svg";
import NSMLogo from "../../img/logos/NSM_logo_hvit.png";
import helseatlasLogo from "../../img/logos/Logo_atlas_hvit.png";
import { imgLoader } from "../../helpers/functions";

import style from "./Footer.module.css";
import Image from "next/image";

export const Footer: React.FC = () => {
  return (
    <footer className={style.footer}>
      <div className={style.footerTop}>
        <div>
          <Link href="/">
            <>
              <Image
                loader={imgLoader}
                src={skdeLogo}
                alt="skde logo"
                width={99}
                height={40}
              />
            </>
          </Link>
        </div>
        <div className={style.skdeContact}>
          <strong>Telefon:</strong>
          <a href="tel:77755800"> 777 55 800</a>
          <br />
          <strong>E-post:</strong>
          <a href="mailto:postmottak@helse-nord.no">
            {" "}
            postmottak@helse-nord.no
          </a>
          <br />
          <strong>Webredakt√∏r:</strong>
          <a href="mailto:webmaster@skde.no"> webmaster@skde.no</a>
          <br />
        </div>
        <div className={style.skdePrivacy}>
          <Link title="Link til kontakt SKDE" href={`/kontakt`}>
            Kontakt
          </Link>
          <br />
          <Link title="Link til personvern" href={`/personvern`}>
            Personvern
          </Link>
          <Link
            href="/informasjonskapsler"
            title="Link til informasjon om informasjonskapsler"
          >
            Informasjonskapsler
          </Link>
        </div>
      </div>
      <div className={style.footerBottom}>
        <div>
          <a title="Link til Helse Nord RHF" href="https://www.helse-nord.no">
            <Image
              loader={imgLoader}
              src={helseNordLogo}
              alt="Helse Nord logo"
              width={180}
              height={40}
            />
          </a>
        </div>
        <div>
          <a
            title="Link til kvalitetsregistre"
            href="https://www.kvalitetsregistre.no"
          >
            <Image
              loader={imgLoader}
              src={NSMLogo}
              alt="NSM logo"
              width={287}
              height={32}
            />
          </a>
        </div>
        <div>
          <Link title="Link til helseatlas" href="/helseatlas">
            <>
              <Image
                loader={imgLoader}
                src={helseatlasLogo}
                alt="Helseatlas logo"
                width={146}
                height={40}
              />
            </>
          </Link>
        </div>
      </div>
    </footer>
  );
};
