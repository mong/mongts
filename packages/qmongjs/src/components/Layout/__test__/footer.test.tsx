import { render } from "@testing-library/react";

import { Footer } from "../Footer";

it("Footer renders", async () => {
  const { container } = render(<Footer />);

  expect(container).toMatchSnapshot();
});
