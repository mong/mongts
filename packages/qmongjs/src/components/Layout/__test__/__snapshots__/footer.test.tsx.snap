// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders 1`] = `
<div>
  <footer
    class="footer"
  >
    <div
      class="footerTop"
    >
      <div>
        <a
          href="/"
        >
          <img
            alt="skde logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="SKDE_hvit_lys.png?w=256&q=75"
            srcset="SKDE_hvit_lys.png?w=128&q=75 1x, SKDE_hvit_lys.png?w=256&q=75 2x"
            style="color: transparent;"
            width="99"
          />
        </a>
      </div>
      <div
        class="skdeContact"
      >
        <strong>
          Telefon:
        </strong>
        <a
          href="tel:77755800"
        >
           777 55 800
        </a>
        <br />
        <strong>
          E-post:
        </strong>
        <a
          href="mailto:postmottak@helse-nord.no"
        >
           
          postmottak@helse-nord.no
        </a>
        <br />
        <strong>
          Webredakt√∏r:
        </strong>
        <a
          href="mailto:webmaster@skde.no"
        >
           webmaster@skde.no
        </a>
        <br />
      </div>
      <div
        class="skdePrivacy"
      >
        <a
          href="/kontakt"
          title="Link til kontakt SKDE"
        >
          Kontakt
        </a>
        <br />
        <a
          href="/personvern"
          title="Link til personvern"
        >
          Personvern
        </a>
        <br />
        <a
          href="/informasjonskapsler"
          title="Link til informasjon om informasjonskapsler"
        >
          Informasjonskapsler
        </a>
      </div>
    </div>
    <div
      class="footerBottom"
    >
      <div>
        <a
          href="https://www.helse-nord.no"
          title="Link til Helse Nord RHF"
        >
          <img
            alt="Helse Nord logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="hf_nord-white.svg?w=384&q=75"
            srcset="hf_nord-white.svg?w=256&q=75 1x, hf_nord-white.svg?w=384&q=75 2x"
            style="color: transparent;"
            width="180"
          />
        </a>
      </div>
      <div>
        <a
          href="https://www.kvalitetsregistre.no"
          title="Link til kvalitetsregistre"
        >
          <img
            alt="NSM logo"
            data-nimg="1"
            decoding="async"
            height="32"
            loading="lazy"
            src="NSM_logo_hvit.png?w=640&q=75"
            srcset="NSM_logo_hvit.png?w=384&q=75 1x, NSM_logo_hvit.png?w=640&q=75 2x"
            style="color: transparent;"
            width="287"
          />
        </a>
      </div>
      <div>
        <a
          href="/helseatlas"
          title="Link til helseatlas"
        >
          <img
            alt="Helseatlas logo"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="Logo_atlas_hvit.png?w=384&q=75"
            srcset="Logo_atlas_hvit.png?w=256&q=75 1x, Logo_atlas_hvit.png?w=384&q=75 2x"
            style="color: transparent;"
            width="146"
          />
        </a>
      </div>
    </div>
  </footer>
</div>
`;
